node.mjs

```diff
@@ -54,44 +54,50 @@ let modPromise = null;
 let initialised = false;
 
 /**
- * Loads the WASM module synchronously
+ * Loads and instantiates the WASM module synchronously
  *
- * It will throw if there is an attempt to laod the module asynchronously running
+ * It will throw if there is an attempt to load the module asynchronously running
  *
- * @returns {WebAssembly.Module}
+ * @returns {typeof import("./pkg/matrix_sdk_crypto_wasm_bg.wasm.d")}
  */
 function loadModuleSync() {
-    if (modPromise) throw new Error("The WASM module is being loadded asynchronously but hasn't finished");
+    if (modPromise) throw new Error("The WASM module is being loaded asynchronously but hasn't finished");
     const bytes = readFileSync(filename);
-    return new WebAssembly.Module(bytes);
+    const mod = new WebAssembly.Module(bytes);
+
+    const instance = new WebAssembly.Instance(mod, {
+        // @ts-expect-error: The bindings don't exactly match the 'ExportValue' type
+        "./matrix_sdk_crypto_wasm_bg.js": bindings,
+    });
+
+    // @ts-expect-error: Typescript doesn't know what the instance exports exactly
+    return initInstance(instance);
 }
 
 /**
- * Loads the WASM module asynchronously
+ * Loads and instantiates the WASM module asynchronously
  *
- * @returns {Promise<WebAssembly.Module>}
+ * @returns {Promise<typeof import("./pkg/matrix_sdk_crypto_wasm_bg.wasm.d")>}
  */
-async function loadModule() {
+async function loadModuleAsync() {
     const bytes = await readFile(filename);
-    return await WebAssembly.compile(bytes);
+    const instance = await WebAssembly.instantiate(bytes, {
+        // @ts-expect-error: The bindings don't exactly match the 'ExportValue' type
+        "./matrix_sdk_crypto_wasm_bg.js": bindings,
+    });
+
+    // @ts-expect-error: Typescript doesn't know what the instance exports exactly
+    return initInstance(instance);
 }
 
 /**
  * Initializes the WASM module and returns the exports from the WASM module.
  *
- * @param {WebAssembly.Module} mod
+ * @param {WebAssembly.Instance} instance
  * @returns {typeof import("./pkg/matrix_sdk_crypto_wasm_bg.wasm.d")}
  */
-function initInstance(mod) {
+function initInstance(instance) {
     if (initialised) throw new Error("initInstance called twice");
-
-    /** @type {{exports: typeof import("./pkg/matrix_sdk_crypto_wasm_bg.wasm.d")}} */
-    // @ts-expect-error: Typescript doesn't know what the instance exports exactly
-    const instance = new WebAssembly.Instance(mod, {
-        // @ts-expect-error: The bindings don't exactly match the 'ExportValue' type
-        "./matrix_sdk_crypto_wasm_bg.js": bindings,
-    });
-
     bindings.__wbg_set_wasm(instance.exports);
     instance.exports.__wbindgen_start();
```

Comment:
Because we don't type cast of the exports, this will result to an error
```suggestion
    // @ts-expect-error: Typescript doesn't know what the module exports are
    instance.exports.__wbindgen_start();
```